### PR TITLE
#70: pull in pressiolinalg pypi package

### LINKED ISSUE #70: Linear algebra libraries: mpi4py light library and pressio-tools full library.
To add concrete implementations that can be run parallel (row distributed memory), we need a variety of basic algebra and linear algebra functions. 
We want to be able to create a new library that has two usages:
1. a "light mode": 
  - installable via `pip install ...` and has pure python dependencies (if any)
  - supports some of the basic functions so that a user can utilize most of the rom-tools library without headaches

2. a "heavy more performant mode": 
  - uses bindings to Trilinos building on top of what we have done already in https://github.com/Pressio/pressio-tools

NOTE: All functions should target **row distributed parallelization** 

**For the python-only library mode 1 we will support**
- min max of a distributed vector
- A^T B, where A and B are both row distributed
- Hacks to do the SVD and orthogonalization via method of snapshots
- Stretch/unlikely: svd, qr, basic linear solve.

**For the more heavy library mode 2 we will support**
- min max of a distributed vector
- A^T B, where A and B are both row distributed
- Hacks to do the SVD and orthogonalization via method of snapshots
- svd
- qr
- linear solve

### PR DESCRIPTION
**Pull request by cwschilly**
Fixes #70 

Related to [pressio-linalg PR 8](https://github.com/Pressio/pressio-linalg/pull/8)

---

### Changes

`pressio-linalg` is now a required dependency.

- If you pass an MPI communicator to any of the functions, they will run in parallel. Otherwise, the library defaults to running serial.

---

### Usage

```python
import pressiolinalg.linalg [as pla]
```

_Note that there is no hyphen for the import._

---

### Current API
```python
pla.max(a, axis=None, comm=None)
pla.min(a, axis=None, comm=None)
pla.mean(a, axis=None, dtype=None, comm=None)
pla.std(a, axis=None, dtype=None, ddof=0, comm=None)
pla.product(flag1, flag2, alpha, A, B, beta, C, comm=None)
pla.thin_svd(snapshots, comm=None, method='auto')
```

These are all subject to change and will be documented in [this issue](https://github.com/Pressio/pressio-linalg/issues/2).

### CODE REVIEW

**Comment by cwschilly** [comment]
@fnrizzi @eparish1 This PR is ready for review.

I added a unit test for the new package, but it will likey become unnecessary once we begin to incorporate the package into source code (since existing tests would then catch any problems).

`pressio-linalg` also has its own testing suite already.

That being said, should I keep the test in or drop it from the PR?

**Comment by fnrizzi** [comment]
> That being said, should I keep the test in or drop it from the PR?

the linalg library should have its own testsuite. since that is the case, we should remove tests related to it from this repo. 
The romtools repo should only use the linalg package, so just assuming it works.

**Comment by fnrizzi** [comment]
@cwschilly can you also please replace in romtools all functions to call the linalg library functions?

**Comment by eparish1** [comment]
@cwschilly @fnrizzi I agree about Francesco's comments on the tests. But this looks good. I'm fine to replace the calls to linalg in this PR or another one, whatever is easiest.

**Comment by cwschilly** [comment]
@fnrizzi @eparish1 I've added in `pla` where applicable--there's a call to `np.max` in the Scaler that I couldn't update yet since the `pla.max` only works on rank-1 arrays.

**Comment by fnrizzi** [comment]
> @fnrizzi @eparish1 I've added in `pla` where applicable--there's a call to `np.max` in the Scaler that I couldn't update yet since the `pla.max` only works on rank-1 arrays.

please add this to linalg then

**Comment by cwschilly** [comment]
@fnrizzi @eparish1 All calls to `np.max`, `np.min` and SVD (method of snapshots) have been replaced

**Review by eparish1 — APPROVED** [review]
@cwschilly This is wonderful, thanks for doing all of this. This looks great by me. @fnrizzi good by you?

File: romtools/vector_space/utils/scaler.py [hunk 1]
```diff
@@ -214,7 +215,7 @@ def initialize_scalings(self, data_tensor: np.ndarray) -> None:
         for i in range(n_var):
             ith_var = data_tensor[i]
             if self.__scaling_type == 'max_abs':
-                var_scale = np.max(abs(ith_var))
+                var_scale = pla.max(abs(ith_var))
             elif self.__scaling_type == 'mean_abs':
                 var_scale = np.mean(abs(ith_var))
```
Comment (fnrizzi) [inline romtools/vector_space/utils/scaler.py]: why are we not using the pla for the mean?

Comment (cwschilly) [reply romtools/vector_space/utils/scaler.py]: No reason--I'll add that in ASAP

**Comment by eparish1** [comment]
@cwschilly @fnrizzi  you both are tracking where things are at here better than me. Could one of you do the merge when things are ready? Thanks!

**Comment by cwschilly** [comment]
@fnrizzi @eparish1 This PR should be ready to go now

**Review by fnrizzi — APPROVED** [review]

**Pull request merged**